resources/aws_ec2.go

```diff
@@ -66,17 +87,31 @@ var AwsRegions = []string{
 // AWS credentials must be present in ~/.aws - For detailed instructions see
 // http://docs.aws.amazon.com/cli/latest/userguide/cli-config-files.html
 type AwsEc2Res struct {
-	BaseRes `yaml:",inline"`
-	State   string `yaml:"state"`   // state: running, stopped, terminated
-	Region  string `yaml:"region"`  // region must match an element of AwsRegions
-	Type    string `yaml:"type"`    // type of ec2 instance, ie. t2.micro
-	ImageID string `yaml:"imageid"` // imageid must be available on the chosen region
+	BaseRes     `yaml:",inline"`
+	State       string `yaml:"state"`       // state: running, stopped, terminated
+	Region      string `yaml:"region"`      // region must match an element of AwsRegions
+	Type        string `yaml:"type"`        // type of ec2 instance, ie. t2.micro
+	ImageID     string `yaml:"imageid"`     // imageid must be available on the chosen region
+	SnsEndpoint string `yaml:"snsendpoint"` // the public url endpoint to subscribe to events, ie. http://server:12345/
+	SnsPort     uint16 `yaml:"snsport"`     // the local port connected to the above url, ie. 23456
 	// UserData is used to run bash and cloud-init commands on first launch.
 	// See http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html
 	// for documantation and examples.
 	UserData string `yaml:"userdata"`
 
-	client *ec2.EC2 // client session for AWS API calls
+	client    *ec2.EC2              // client session for AWS API calls
+	snsClient *sns.SNS              // client session for SNS API calls
+	cweClient *cwe.CloudWatchEvents // client session for SNS API calls
+	srv       *http.Server          // http server for SNS endpont
+	awsChan   chan *chanStruct      // channel for aws events
+	closeChan chan struct{}         // a channel to receive signals shut down
+	topicArn  string                // the topicArn we need to delete the topic when we're shutting down
+}
+
+// a struct for passing data around watch
```

Comment:
please follow the doc convention even though this struct is private. it also should be a sentence and end with a .